polaris.shopify.com/src/components/Code/Code.tsx

```diff
@@ -0,0 +1,109 @@
+import Tooltip from "../Tooltip";
+import Prism from "prismjs";
+import { useCopyToClipboard } from "../../utils/hooks";
+import styles from "./Code.module.scss";
+import { Tab } from "@headlessui/react";
+import Image from "../Image";
+import { useState } from "react";
+
+interface Props {
+  code:
+    | {
+        title: string;
+        code: string;
+      }
+    | {
+        title: string;
+        code: string;
+      }[];
+}
+
+function Code({ code }: Props) {
+  const [selectedIndex, setSelectedIndex] = useState(0);
+
+  return (
+    <div className={styles.Code}>
+      {Array.isArray(code) ? (
+        <Tab.Group selectedIndex={selectedIndex} onChange={setSelectedIndex}>
+          <div className={styles.TopBar}>
+            <Tab.List className={styles.Tabs}>
+              {code.map(({ title }) => (
+                <Tab key={title} className={styles.Tab}>
+                  {title}
+                </Tab>
+              ))}
+            </Tab.List>
+            {code[selectedIndex] && (
+              <CopyButton code={code[selectedIndex].code} />
+            )}
+          </div>
+
+          <Tab.Panels>
+            {code.map(({ title, code }) => (
+              <Tab.Panel key={title}>
+                <HighlightedCode code={code} />
+              </Tab.Panel>
+            ))}
+          </Tab.Panels>
+        </Tab.Group>
+      ) : (
+        <>
+          <div className={styles.TopBar}>
+            <div className={styles.Tabs}>
+              <div className={styles.Tab}>{code.title}</div>
+            </div>
+            <CopyButton code={code.code} />
+          </div>
+          <HighlightedCode code={code.code} />
+        </>
+      )}
+    </div>
+  );
+}
+
+function HighlightedCode({ code }: { code: string }) {
+  return (
+    <pre>
+      <code
+        className={styles.ActualCode}
+        dangerouslySetInnerHTML={{
+          __html: Prism.highlight(
+            code,
+            Prism.languages.javascript,
+            "javasript"
```

Comment:
Found a typo!

also with Prism, it’s best practice to set Prism.manual = true somewhere at the top of this file. It shouldn’t make a difference here because you haven’t used the classes on the containing element that prism searches for to replace the content, but, that can get funky. Prism.manual is also just a flag for the internals of prism not to run unless called so it’s a minor optimization but worth it :)

Reply:
Fixed the typo here b3e14b733119519952cc1fbf20541c5ad43022d6

Are you able to write up an issue around what `Prism.manual = true` would change or enhance with the code highlighting? I am taking a look and am not too sure I understand if it's needed.